cvat-ui/src/components/md-guide/guide-page.tsx

```diff
@@ -0,0 +1,191 @@
+// Copyright (C) 2023 CVAT.ai Corporation
+//
+// SPDX-License-Identifier: MIT
+
+import './styles.scss';
+
+import React, {
+    useState, useEffect, useRef, useCallback,
+} from 'react';
+import { useLocation, useParams } from 'react-router';
+import { Row, Col } from 'antd/lib/grid';
+import notification from 'antd/lib/notification';
+import Button from 'antd/lib/button';
+import Space from 'antd/lib/space';
+import MDEditor, { commands } from '@uiw/react-md-editor';
+
+import {
+    getCore, Task, Project, AnnotationGuide,
+} from 'cvat-core-wrapper';
+import { useIsMounted } from 'utils/hooks';
+import CVATLoadingSpinner from 'components/common/loading-spinner';
+import GoBackButton from 'components/common/go-back-button';
+
+const core = getCore();
+
+function GuidePage(): JSX.Element {
+    const mdEditorRef = useRef<typeof MDEditor & { commandOrchestrator: commands.TextAreaCommandOrchestrator }>(null);
+    const location = useLocation();
+    const isMounted = useIsMounted();
+    const [value, setValue] = useState('');
+    const instanceType = location.pathname.includes('projects') ? 'project' : 'task';
+    const id = +useParams<{ id: string }>().id;
+    const [guide, setGuide] = useState<AnnotationGuide>(
+        new AnnotationGuide({
+            ...(instanceType === 'project' ? { project_id: id } : { task_id: id }),
+            markdown: value,
+        }),
+    );
+    const [fetching, setFetching] = useState(false);
+
+    useEffect(() => {
+        setFetching(true);
+        const promise = instanceType === 'project' ? core.projects.get({ id }) : core.tasks.get({ id });
+        promise.then(([instance]: [Task | Project]) => (
+            instance.guide()
+        )).then((guideInstance: AnnotationGuide | null) => {
+            if (guideInstance) {
+                return (async () => guideInstance)();
+            }
```

Comment:
Will async callback work instead of `return (async () => guideInstance)();` ?
```
)).then(async (guideInstance: AnnotationGuide | null) => {
            if (guideInstance) {
                return  guideInstance;
            }
```

Reply:
Works, done